native-sql-engine/cpp/src/operators/columnar_to_row_converter.h

```diff
@@ -25,35 +25,39 @@
 #include <arrow/type.h>
 #include <arrow/util/bit_util.h>
 
+#include <boost/align.hpp>
+
 #include "gandiva/decimal_type_util.h"
 
 namespace sparkcolumnarplugin {
 namespace columnartorow {
 
 class ColumnarToRowConverter {
  public:
-  ColumnarToRowConverter(std::shared_ptr<arrow::RecordBatch> rb,
-                         arrow::MemoryPool* memory_pool)
-      : rb_(rb), memory_pool_(memory_pool) {}
+  ColumnarToRowConverter(arrow::MemoryPool* memory_pool)
+      : memory_pool_(memory_pool), buffer_(nullptr) {}
 
-  arrow::Status Init();
+  arrow::Status Init(const std::shared_ptr<arrow::RecordBatch>& rb);
   arrow::Status Write();
 
   uint8_t* GetBufferAddress() { return buffer_address_; }
-  const std::vector<int64_t>& GetOffsets() { return offsets_; }
-  const std::vector<int64_t>& GetLengths() { return lengths_; }
+  const std::vector<int32_t>& GetOffsets() { return offsets_; }
+  const std::vector<int32_t, boost::alignment::aligned_allocator<int32_t, 32>>&
```

Comment:
I think the second template parameter with value 32 is misspelled.
`boost::alignment::aligned_allocator<int32_t, 32>>`
It should be 8

Reply:
It's used here for 256 bit aligned.

https://github.com/oap-project/gazelle_plugin/blob/47af257666aa28e347792737b0d6c64efb35eb94/native-sql-engine/cpp/src/operators/columnar_to_row_converter_avx512.cc#L161

@zhixingheyi-tian here should be  `_mm256_loada_si256`  not laodu

Reply:
Thanks for your explanation.

Reply:
> It's used here for 256 bit aligned.
> 
> https://github.com/oap-project/gazelle_plugin/blob/47af257666aa28e347792737b0d6c64efb35eb94/native-sql-engine/cpp/src/operators/columnar_to_row_converter_avx512.cc#L161
> 
> @zhixingheyi-tian here should be `_mm256_loada_si256` not laodu

Yes, updated in https://github.com/oap-project/gazelle_plugin/pull/937.
Use _mm256_load_si256 instead of  _mm256_loadu_si256.